sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -491,13 +513,392 @@ function main() {
+> For more information about working with the Confidential Client of MSAL, please refer to [Initialization of MSAL (Node.js)](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/master/lib/msal-node/docs/initialize-confidential-client-application.md).
+> For more information about working with the On Behalf Flow with MSAL, please refer to [On Behalf of Flow](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/master/lib/msal-common/docs/request.md#on-behalf-of-flow).
+
+```ts
+import { TokenCredential, AccessToken } from "@azure/core-auth";
+import * as msalNode from "@azure/msal-node";
+
+class OnBehalfOfCredential implements TokenCredential {
+  private confidentialApp: msalNode.ConfidentialClientApplication;
+
+  constructor (
+    private clientId: string,
+    private clientSecret: string,
+    private userAccessToken: string
+  ) {
+    this.confidentialApp = new msalNode.ConfidentialClientApplication({
+      auth: {
+        clientId,
+        clientSecret
+      }
+    });
+  }
+  async getToken(scopes: string | string[]): Promise<AccessToken> {
+    const result = await this.confidentialApp.acquireTokenOnBehalfOf({
+      scopes: Array.isArray(scopes) ? scopes : [scopes],
+      oboAssertion: this.userAccessToken
+    });
+    return {
+      token: result.accessToken,
+      expiresOnTimestamp: result.expiresOn.getTime()
+    };
+  }
+}
+```
+
+The following example shows an how the `OnBehalfOfCredential` could be used to authenticate a `SecretClient`.
+
+```ts
+import { SecretClient } from "@azure/keyvault-secrets";
+
+async function main() {
+  const oboCredential = new OnBehalfOfCredential(clientId, clientSecret, userAccessToken);
+
+  const client = new SecretClient("https://myvault.vault.azure.net/", oboCredential);
+}
+```
+
+#### Authenticating with the @azure/msal-browser Public Client
+
+While `@azure/identity` provides some browser support, for users that need the complete set of features offered by `@azure/msal-browser`, it is possible to implement a `TokenCredential` on top of MSAL's public API for the browsers.
+
+You'll need to install the [@azure/msal-browser][msal_browser_npm] and the the [@azure/core-auth][core_auth] package for this sample.
+
+> For more information about MSAL for browsers, please refer to [the README of the `@azure/msal-browser` package][msal_browser_readme].
+
+For this example, we will define a `BrowserCredential` class that has a `getToken` method (which will use the Silent Authentication flow, retrieving the account from memory, as we want to prevent unnecessary redirections), and also a `prepare()` method that will try either to check if the account has previously authenticated, or to parse the redirection URI values if present, also a `hasAuthenticated` method which can be used to know if the authentication has taken place, and finally a `loginRedirect` method, which if called will trigger the authentication via redirection.
```

Comment:
```suggestion
For this example, you'll define a `BrowserCredential` class with the following methods:

- `getToken` &mdash; Will use the Silent Authentication flow, retrieving the account from memory to prevent unnecessary redirections.
- `prepare` &mdash; Will try either to check if the account has previously authenticated or to parse the redirection URI values if present.
- `hasAuthenticated` &mdash; Can be used to determine if the authentication has taken place.
- `loginRedirect` &mdash; If called, this method triggers the authentication via redirection.
```